subprojects/plugin-development/src/main/java/org/gradle/plugin/devel/plugins/JavaGradlePluginPlugin.java

```diff
@@ -152,7 +152,17 @@ private void registerPlugins(Project project, GradlePluginDevelopmentExtension e
 
     private void applyDependencies(Project project) {
         DependencyHandler dependencies = project.getDependencies();
-        dependencies.add(API_CONFIGURATION, dependencies.gradleApi());
+        String apiVersion = System.getProperty("org.gradle.api.version");
+        if (apiVersion == null || apiVersion.isEmpty()) {
+            dependencies.add(API_CONFIGURATION, dependencies.gradleApi());
+        } else {
+            String repositoryUrl = System.getProperty("gradle.api.repository.url", "https://repo.gradle.org/gradle/libs-releases");
+            project.getRepositories().maven(repository -> {
+                repository.setName("gradleApiRepository");
+                repository.setUrl(repositoryUrl);
+            });
+            dependencies.add(API_CONFIGURATION, dependencies.gradleApi(apiVersion));
+        }
```

Comment:
I think it would be good if the user could declare the API version they want to use via a `gradleApi("6.9")` call in the build script of the plugin. The above mechanism does not allow that.

Reply:
Yes, this can be improved. Note that with this solution you can already specify a different version of the gradle API by adding a constraint to the configuration, since the `gradleApi(version)` is a regular dependency and participates in dependency resolution. For the final solution, I am not sure what the right thing from dependency management is. `defaultDependencies` sounds it may be something in the right direction.